### DISCUSSION TITLE
Public Server Utilization/Performance Peak Hours?

### DISCUSSION THREAD

**Discussion by itemforty** [Lurkars/gloomhavensecretariat#215] [Q&A]
Our group met last Friday, 3/7/2023, to play at our favorite bar like we've done many times in the past. We played a 2 hour scenario with GHS over the public server, 4 devices using the app, with a 5th device streaming to a projector with a Chromecast. Everything was going great, but during the middle of our second, longer scenario, we started to notice painfully slow lag between the client and server (3-5+ seconds). This started to happen roughly around 8:30PM CST/Texas, when the bar started to get a little busier, although we were using the employee WIFI, which wouldn't get as saturated. 

At the time we guessed that maybe the public server was getting rattled by the increased traffic from other people having game night. Is that the most likely culprit? Or is there any other improvement I can make to our setup or connection to make the experience better? 

Thanks in advance for any help!

**Comment by Lurkars** [comment]
Tough to investigate. I will try to look at the server for errors during this timeframe. Maybe I need to use public server myself more to get feeling, but two big difference: location and timezone. Maybe it's just a location issue with Germany vs Overseas and when our group plays, server maybe not that busy 🤷. If you have time for that, it's always good to maybe export data dump for me for further investigation.

**Comment by itemforty** [comment]
@Lurkars this is my device I use as a client:
[Android Device used as Client - Data Dump.zip](https://github.com/Lurkars/gloomhavensecretariat/files/11203375/Android.Device.used.as.Client.-.Data.Dump.zip)

And this is from the 5th device we use to ChromeCast to a display
[Android Device used for Chromecast - Data Dump.zip](https://github.com/Lurkars/gloomhavensecretariat/files/11203376/Android.Device.used.for.Chromecast.-.Data.Dump.zip)

Hope that helps and thanks for looking!

**Comment by itemforty** [comment]
We played another long session on Saturday, starting around 3:30pm CST (Texas) and ending around 10:30pm CST. Server/client response got to be really slow by the end, and found we could mitigate it a little by closing individual clients out. This time we were playing at my house, all on my WiFi, and I was casting a Chrome tab of GHS from my desktop to a display.

**Reply by Lurkars** [reply]
Okay, so definitely relates to total play time or more activity during later hours for you? Another dump would be nice, it's also possible that you're doing something specific leading to an round-trip update, had some bugs on this before. So that a change triggers all clients to also pull updates to the server etc. If you're somehow familiar with a dev console of a browser, would also be awesome if you're able to take a look in there, if something looks shady or showing errors. By now I still don't have more ideas sadly.

**Comment by itemforty** [comment]
Ok I'll try to get some dumps or find console access when it's happening if that helps. Here's a dump from my phone:

[Android Device used as Client - Data Dump2.zip](https://github.com/Lurkars/gloomhavensecretariat/files/11263593/Android.Device.used.as.Client.-.Data.Dump2.zip)

**Comment by itemforty** [comment]
Just to report back, played again yesterday, my house, 5+ hour session, started to slow down a bit by the end. 

Gonna close this out since I'm gonna try to spin up my own server and go from there.
